src/Standard/TIP-3/1.md

```diff
@@ -0,0 +1,175 @@
+
+## Motivation
+
+A standard interface allows any tokens on Everscale to be re-used by other applications: from wallets to decentralized exchanges.
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Notes
+
+- We choose Standard Interface Detection to expose the interfaces that a TIP3 smart contract supports.
+- This standard does not define the external methods to initiate transfer, mint or burn tokens.
+  Though it defines the methods, which MUST be called on a recipient token wallet or token root during these operations.
+- The rules for a token wallet ownership MUST be defined in a child standards.
+- A `-1` offset is added to some function IDs derivations, so the preimage of the hash cannot be known, further reducing the chances of a possible collisions.
+
+### Token root
+
+#### Name
+
+Returns the name of the token - e.g. "MyToken".
+
+```solidity
+function name() public view responsible returns (string);
+```
+
+#### Symbol
+
+Returns the symbol of the token. E.g. “HIX”.
+
+```solidity
+function symbol() public view responsible returns (string);
+```
+
+#### Decimals
+
+Returns the number of decimals the token uses - e.g. 8, means to divide the token amount by 100000000 to get its user representation.
+
+```solidity
+function decimals() public view responsible returns (uint8);
+```
+
+#### Total supply
+
+Returns the total token supply.
+
+```solidity
+function totalSupply() public view responsible returns (uint128);
+```
+
+#### Token wallet code
+
+Returns the token wallet code.
+
+```solidity
+function walletCode() public view responsible returns (TvmCell);
+```
+
+#### Accept tokens burn
+
+Does not have a standard signature, but has a standard function ID `0x192B51B1` obtained as `tvm.functionId('acceptBurn(uint128)') - 1`. The `uint128 _value` parameter MUST be first. The function name and the rest of the parameters are not fixed by this standard and can be reinvented for each substandard.
+
+Decreases token total supply by `_value`. The contract MUST check that the sender is a correct token wallet. Before sending this message, caller token wallet MUST decrease its own balance by `_value`. If the mint can't be accepted (e.g. invalid sender), this message MUST be bounced.
+
+Any function from the following snippet is a valid example:
+
+```solidity
+interface TIP3AcceptBurn {
+    function acceptBurn(uint128 _value) functionID(0x192B51B1) external;
+    function acceptBurn2(uint128 _value, uint256 _publicKey, address _owner) functionID(0x192B51B1) external;
+    function acceptBurn3(uint128 _value, TvmCell _meta) functionID(0x192B51B1) external;
+}
+```
+
+#### Standard interface detection
+
+```solidity
+interface TIP3TokenRoot {
+    /// @dev Accept burn function ID `0x192B51B1`
```

Comment:
Declare `acceptBurn` method like `acceptBurn(uint128 value) external functionID(0x192B51B1);`
It will help SDK to decode this method from the message body.